packages/e2e-test-utils/src/open-all-block-inserter-categories.js

```diff
@@ -2,10 +2,10 @@
  * Opens all block inserter categories.
  */
 export async function openAllBlockInserterCategories() {
-	const notOpenCategoryPanels = await page.$$(
-		'.block-editor-inserter__results .components-panel__body:not(.is-opened)'
-	);
-	for ( const categoryPanel of notOpenCategoryPanels ) {
+	const notOppenedCategorySelector = '.block-editor-inserter__results .components-panel__body:not(.is-opened)';
+	let categoryPanel = await page.$( notOppenedCategorySelector );
+	while ( categoryPanel !== null ) {
 		await categoryPanel.click();
+		categoryPanel = await page.$( notOppenedCategorySelector );
```

Comment:
How about we keep the `for of` loop and we wait until it has `is-opened` class applied? I never feel confident about using while loop :)
I'm not sure to be honest, maybe this is okay since it has to `await` and would timeout anyways.

Reply:
I think this approach would still be problematic, because if we used the selector '.block-editor-inserter__results .components-panel__body' in a for of, as soon as we applied the first click, all the other selectors would be invalid, because they reference an element that no longer exists on the dom because of a rerender that happens on click so we would not be able to use the element to wait for a class.

Reply:
Yeah, it's a complex stuff :)